### PR TITLE
Board Testing Suite

### PR DESCRIPTION
**Pull request by kevingloss**
Includes:
- Creating the rspec testing file
- Check methods for:
   - Initialize
   - Cells
   - Valid_Coordinate?
   - Valid_Placement?
   - Placeholder for ship.place

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+require './lib/ship'
+require './lib/cell'
+
+RSpec.describe Board do
+
+  before(:each) do
+    @board = Board.new
+  end
+
+  describe 'initialize' do
```
Comment (joeldenverdev) [inline spec/board_spec.rb:10]: Use # in front of method names for describe blocks. '#initialize'

File: spec/board_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+require './lib/ship'
+require './lib/cell'
+
```
Comment (joeldenverdev) [inline spec/board_spec.rb:3]: add: require './lib/board'

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @board = Board.new
+  end
+
+  describe 'initialize' do
+    it 'is a board' do
+      expect(@board).to be_a(Board)
+    end
+  end
+
+  describe 'cells method' do
```
Comment (joeldenverdev) [inline spec/board_spec.rb:16]: '#cells' to denote a method

File: spec/board_spec.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      expect(@board.valid_placement?(@cruiser, ["A1", "B2", "C3"])).to be(false)
+      expect(@board.valid_placement?(@submarine, ["C2", "D3"])).to be(false)
+    end
+    it 'is a valid placement' do
+      expect(@board.valid_placement?(@cruiser, ["B1", "C1", "D1"])).to be(true)
+      expect(@board.valid_placement?(@submarine, ["A1", "A2"])).to be(true)
+    end
+  end
+
+  describe 'place method' do
```
Comment (joeldenverdev) [inline spec/board_spec.rb:63]: `'#place'`

File: spec/board_spec.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      expect(@board.valid_coordinate?("A22")).to be(false)
+    end
+  end
+
+  before(:each) do
+    @cruiser = Ship.new("Cruiser", 3)
+    @submarine = Ship.new("Submarine", 2)
+  end
+
+  describe 'valid_placement?' do
```
[the remark this answers is not in the document]
Comment (kevingloss) [reply spec/board_spec.rb:41]: Updated

File: spec/board_spec.rb (hunk 4 above)
Comment (kevingloss) [reply spec/board_spec.rb:63]: Updated

File: spec/board_spec.rb (hunk 2 above)
Comment (kevingloss) [reply spec/board_spec.rb:3]: Updated

File: spec/board_spec.rb (hunk 1 above)
Comment (kevingloss) [reply spec/board_spec.rb:10]: Updated

File: spec/board_spec.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    end
+  end
+
+  describe 'cells method' do
+    it 'creates cell objects' do
+      expect(@board.cells).to be_a(Hash)
+      expect(@board.cells.size).to eq(16)
+
+      @board.cells.each do |cell|
+        expect(cell.keys).to be(Cell)
```
[the remark this answers is not in the document]
Comment (joeldenverdev) [reply spec/board_spec.rb]: I agree.  We may not totally know what the test needs to produce yet.

**Pull request merged**